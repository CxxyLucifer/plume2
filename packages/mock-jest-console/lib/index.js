"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var BaseMock = /** @class */ (function () {
    function BaseMock() {
        this._logs = [];
        this.mock();
    }
    BaseMock.prototype.mock = function () { };
    BaseMock.prototype.mockFn = function () {
        var _this = this;
        return jest.fn(function (log) { return _this._logs.push(log); });
    };
    Object.defineProperty(BaseMock.prototype, "logs", {
        get: function () {
            return this._logs;
        },
        enumerable: true,
        configurable: true
    });
    BaseMock.prototype.reset = function () {
        this._logs = [];
    };
    return BaseMock;
}());
exports.BaseMock = BaseMock;
var MockWarn = /** @class */ (function (_super) {
    __extends(MockWarn, _super);
    function MockWarn() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockWarn.prototype.mock = function () {
        console.warn = this.mockFn();
    };
    return MockWarn;
}(BaseMock));
exports.MockWarn = MockWarn;
var MockLog = /** @class */ (function (_super) {
    __extends(MockLog, _super);
    function MockLog() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockLog.prototype.mock = function () {
        console.log = this.mockFn();
    };
    return MockLog;
}(BaseMock));
exports.MockLog = MockLog;
var MockDir = /** @class */ (function (_super) {
    __extends(MockDir, _super);
    function MockDir() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockDir.prototype.mock = function () {
        console.dir = this.mockFn();
    };
    return MockDir;
}(BaseMock));
exports.MockDir = MockDir;
var MockDebug = /** @class */ (function (_super) {
    __extends(MockDebug, _super);
    function MockDebug() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockDebug.prototype.mock = function () {
        console.debug = this.mockFn();
    };
    return MockDebug;
}(BaseMock));
exports.MockDebug = MockDebug;
var MockError = /** @class */ (function (_super) {
    __extends(MockError, _super);
    function MockError() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockError.prototype.mock = function () {
        console.error = this.mockFn();
    };
    return MockError;
}(BaseMock));
exports.MockError = MockError;
var MockTrace = /** @class */ (function (_super) {
    __extends(MockTrace, _super);
    function MockTrace() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockTrace.prototype.mock = function () {
        console.trace = this.mockFn();
    };
    return MockTrace;
}(BaseMock));
exports.MockTrace = MockTrace;
var MockConsole = /** @class */ (function (_super) {
    __extends(MockConsole, _super);
    function MockConsole() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockConsole.prototype.mock = function () {
        console.warn = this.mockFn();
        console.log = this.mockFn();
        console.dir = this.mockFn();
        console.debug = this.mockFn();
        console.error = this.mockFn();
        console.trace = this.mockFn();
    };
    return MockConsole;
}(BaseMock));
exports.MockConsole = MockConsole;
