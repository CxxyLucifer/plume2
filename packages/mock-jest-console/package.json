{
  "name": "mock-jest-console",
  "version": "1.1.1",
  "description": "jest.fn mock console.xxx",
  "main": "./lib/index.js",
  "types": "./lib/typings/index.d.ts",
  "keywords": [
    "mock jest console"
  ],
  "author": "hufeng",
  "license": "MIT"
}
