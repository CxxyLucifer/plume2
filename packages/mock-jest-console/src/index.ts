declare const jest: { fn: Function };

export class BaseMock {
  constructor() {
    this._logs = [];
    this.mock();
  }

  private _logs;

  mock() {}

  mockFn() {
    return jest.fn(log => this._logs.push(log));
  }

  get logs() {
    return this._logs;
  }

  reset() {
    this._logs = [];
  }
}

export class MockWarn extends BaseMock {
  mock() {
    console.warn = this.mockFn();
  }
}

export class MockLog extends BaseMock {
  mock() {
    console.log = this.mockFn();
  }
}

export class MockDir extends BaseMock {
  mock() {
    console.dir = this.mockFn();
  }
}

export class MockDebug extends BaseMock {
  mock() {
    console.debug = this.mockFn();
  }
}

export class MockError extends BaseMock {
  mock() {
    console.error = this.mockFn();
  }
}

export class MockTrace extends BaseMock {
  mock() {
    console.trace = this.mockFn();
  }
}

export class MockGroup extends BaseMock {
  mock() {
    console.group = this.mockFn();
  }
}

export class MockGroupCollapsed extends BaseMock {
  mock() {
    console.groupCollapsed = this.mockFn();
  }
}

export class MockTime extends BaseMock {
  mock() {
    console.time = this.mockFn();
  }
}

export class MockConsole extends BaseMock {
  mock() {
    console.warn = this.mockFn();
    console.log = this.mockFn();
    console.dir = this.mockFn();
    console.debug = this.mockFn();
    console.error = this.mockFn();
    console.trace = this.mockFn();
    console.group = this.mockFn();
    console.groupCollapsed = this.mockFn();
    console.time = this.mockFn();
  }
}
