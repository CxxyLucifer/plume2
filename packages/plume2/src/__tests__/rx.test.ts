import { fromJS } from 'immutable';
import { MockConsole, MockLog } from 'mock-jest-console';
import { Action, Actor, IMap, QL, RL, Store } from '../index';

class HelloActor extends Actor {
  defaultState() {
    return { mott: 'plume2' };
  }

  @Action('change')
  change(state: IMap) {
    return state.set('mott', 'Build tools for human');
  }
}

const helloQL = QL('helloQL', ['mott', mott => mott]);

const helloRL = RL('helloRL', [
  'mott',
  helloQL,
  (mott, mottQL) => {
    //side-effect
    //webapi.update();
    expect(mott).toEqual('Build tools for human');
    expect(mottQL).toEqual('Build tools for human');
  }
]);

class AppStore extends Store {
  bindActor() {
    return [HelloActor];
  }

  bindRx() {
    return { helloRL };
  }
}

describe('reactive rx test suite', () => {
  it('test helloRl', () => {
    const mockLog = new MockLog();
    new AppStore({
      debug: true
    });
    expect(mockLog.logs).toMatchSnapshot();
  });

  it('test helloRL dispatch one time', () => {
    const mockLog = new MockLog();
    const store = new AppStore({
      debug: true
    });
    store.dispatch('change');
    expect(mockLog.logs).toMatchSnapshot();
  });

  it('test helloRL dispatch second time', () => {
    const mockLog = new MockLog();
    const store = new AppStore({
      debug: true
    });
    store.dispatch('change');
    store.dispatch('change');
    expect(mockLog.logs).toMatchSnapshot();
  });

  it('test helloRL dispatch transaction', () => {
    const mockConsole = new MockConsole();
    const store = new AppStore({ debug: true });
    store.transaction(() => {
      store.dispatch('change');
      store.dispatch('change');
      store.dispatch(
        'c',
        fromJS({
          id: 1,
          name: 'plume2',
          addr: {
            province: 'jiangsu',
            city: 'nanjing'
          }
        })
      );
    });
    expect(mockConsole.logs).toMatchSnapshot();
  });
});
