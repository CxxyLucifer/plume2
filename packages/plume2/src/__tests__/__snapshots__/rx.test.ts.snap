// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reactive rx test suite test helloRL dispatch one time 1`] = `
Array [
  "Store init ðŸš€",
  "Actor:reduce HelloActor",
  "parse:rl -> helloRL",
  "ðŸ”¥:tracing: QL(helloQL)",
  "dep:mott, cache:false, value:\\"plume2\\"",
  "QL(helloQL)|> \\"plume2\\"",
  "store dispatch => 'change'",
  "HelloActor receive => 'change'",
]
`;

exports[`reactive rx test suite test helloRL dispatch second time 1`] = `
Array [
  "Store init ðŸš€",
  "Actor:reduce HelloActor",
  "parse:rl -> helloRL",
  "ðŸ”¥:tracing: QL(helloQL)",
  "dep:mott, cache:false, value:\\"plume2\\"",
  "QL(helloQL)|> \\"plume2\\"",
  "store dispatch => 'change'",
  "HelloActor receive => 'change'",
  "store dispatch => 'change'",
  "HelloActor receive => 'change'",
]
`;

exports[`reactive rx test suite test helloRL dispatch transaction 1`] = `
Array [
  "Store init ðŸš€",
  "Actor:reduce HelloActor",
  "parse:rl -> helloRL",
  "ðŸ”¥:tracing: QL(helloQL)",
  "dep:mott, cache:false, value:\\"plume2\\"",
  "QL(helloQL)|> \\"plume2\\"",
  "::::::::::::::::ðŸš€ open new transaction ðŸš€::::::::::::::::::",
  "store dispatch => 'change'",
  "HelloActor receive => 'change'",
  "store dispatch => 'change'",
  "HelloActor receive => 'change'",
  "store dispatch => 'c'",
  "params|> immutable[yes] ",
  Object {
    "addr": Object {
      "city": "nanjing",
      "province": "jiangsu",
    },
    "id": 1,
    "name": "plume2",
  },
]
`;

exports[`reactive rx test suite test helloRl 1`] = `
Array [
  "Store init ðŸš€",
  "Actor:reduce HelloActor",
  "parse:rl -> helloRL",
  "ðŸ”¥:tracing: QL(helloQL)",
  "dep:mott, cache:false, value:\\"plume2\\"",
  "QL(helloQL)|> \\"plume2\\"",
]
`;
